### PR TITLE
Allow admin/user to filter by state for Pods, Clusters and Hosts

### PR DESCRIPTION
**Pull request by kishankavala**
### Description

This PR allows admin/user to filter by state for Pods, Clusters and Hosts. This is part of https://github.com/apache/cloudstack/issues/7366 .

Filters available on UI : `Enabled`, `Disabled`

Pods, Clusters: Filter by allocation state
Hosts: Filter by resource state

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?
Tested UI changes manually

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
@blueorangutan package

File: ui/src/config/section/infra/hosts.js [hunk 1]
```diff
@@ -24,6 +24,10 @@ export default {
   icon: 'desktop-outlined',
   permission: ['listHostsMetrics'],
   resourceType: 'Host',
+  filters: () => {
+    const filters = ['enabled', 'disabled']
```
Comment (rohityadavcloud) [inline ui/src/config/section/infra/hosts.js:28]: @kishankavala could you explore other states for the resources. For example as an admin I only want to see hosts in Alert, Maintenance and other useful states? Pl do the same for other resources (pods, cluster) in this PR

**Comment by rohityadavcloud** [comment]
cc @vishesh92

**Comment by vishesh92** [comment]
Checking this PR in a bit

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

**Comment by rohityadavcloud** [comment]
@blueorangutan package

**Review by rohityadavcloud — CHANGES_REQUESTED** [review]
Overall LGTM. Tested UI changes, I think the host state list could be added to have all the possible states for hosts here, which may be useful for the operator such as Alert etc. Also for host, the agent/state may be a more useful state to filter on than the resource state. The agent/host status can be Up/Down/Alert... see https://github.com/apache/cloudstack/blob/main/api/src/main/java/com/cloud/host/Status.java#L26

File: ui/src/config/section/infra/hosts.js (hunk 1 above)
Comment (shwstppr) [reply ui/src/config/section/infra/hosts.js:28]: @rohityadavcloud cc @kishankavala I guess this PR adds filter for host `resourcestate` which can only be,
```
Creating,
Enabled,
Disabled,
ErrorInPrepareForMaintenance,
PrepareForMaintenance,
ErrorInMaintenance,
Maintenance,
Error,
Degraded
```

Not sure if we have filtering for host state as well which can be host `status` as well which can be,

```
Creating
Connecting
Up
Down
Disconnected
Alert
Removed
Error
Rebalancing
Unknown
```

Comment (rohityadavcloud) [reply ui/src/config/section/infra/hosts.js:28]: @kishankavala any update on this? Thanks.

**Comment by rohityadavcloud** [comment]
We're pending one change, otherwise this can be merged. @kishankavala could you address the review comments, thanks.

**Comment by kishankavala** [comment]
Making changes to add suggested host stated

**Comment by kishankavala** [comment]
Cluster has AllocationState and ManagedState. Filter uses AllocationState. ManagedState  can be added if required
 - AllocationState {
        Enabled, Disabled;
    };
- ManagedState {
        Managed, PrepareUnmanaged, Unmanaged, PrepareUnmanagedError
    };

**Comment by kishankavala** [comment]
Pod has only allocation state and is added to the filter.
- AllocationState {
        Enabled, Disabled;
    };

**Comment by kishankavala** [comment]
Host has State and ResourceState. Its a long list even after removing transition states. 
- State {
    Up,
    Down,
    Disconnected,
    Alert,
    Removed,
    Error,
    Unknown
}

- ResourceState {
    Enabled,
    Disabled,
    ErrorInPrepareForMaintenance,
    PrepareForMaintenance,
    ErrorInMaintenance,
    Maintenance,
    Error,
    Degraded
}

Added the Up, Down, Alert, Enabled, Disabled, Maintenance in the filter

**Comment by kishankavala** [comment]
@blueorangutan ui

File: ui/src/views/AutogenView.vue [hunk 2]
```diff
@@ -1573,6 +1575,20 @@ export default {
...
+        } else {
+          query.allocationstate = filter
+        }
+      } else if (['host'].includes(this.$route.name)) {
+        if (filter === 'all') {
+          delete query.resourcestate
+        } else if (['up', 'down', 'alert'].includes(filter)) {
+          query.state = filter
+        } else {
+          query.resourcestate = filter
```
Comment (rohityadavcloud) [inline ui/src/views/AutogenView.vue:1590]:
```suggestion
          delete query.resourcestate
          delete query.state
        } else if (['up', 'down', 'alert'].includes(filter)) {
          delete query.resourcestate
          query.state = filter
        } else {
          delete query.state
          query.resourcestate = filter
```

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan ui

**Review by shwstppr — APPROVED** [review]
LGTM. Testing filtering resources on qa server

**Pull request merged**